### PR TITLE
feat: add spot rebalance recommendation handling for aws

### PR DESCRIPTION
**Pull request by r0kas**

### CODE REVIEW

File: handler/handler_test.go [hunk 1]
```diff
@@ -160,12 +160,50 @@ func TestRunLoop(t *testing.T) {
 			r.Equal(3, mothershipCalls)
 		}()
 	})
+
+	t.Run("handle successful mock rebalabnce recommendation", func(t *testing.T) {
```
Comment (aldor007) [inline handler/handler_test.go:164]:
```suggestion
	t.Run("handle successful mock rebalance recommendation", func(t *testing.T) {
```

File: handler/handler.go [hunk 2]
```diff
@@ -80,11 +88,26 @@ func (g *SpotHandler) Run(ctx context.Context) error {
 					// Stop after ACK.
 					t.Stop()
 				}
+
+				if !rebalanceRecommendationSent {
+					rebalanceRecommendation, err := g.metadataChecker.CheckRebalanceRecommendation(ctx)
+					if err != nil {
+						return err
```
Comment (aldor007) [inline handler/handler.go:95]: Why do we return in this method at all? maybe we should just log the error? We will retry after `pollWaitInterval`, =

**Review by aldor007 — APPROVED** [review]

File: handler/handler.go (hunk 2 above)
Comment (r0kas) [reply handler/handler.go:95]: That's what is done when it's out of the function
```			if err != nil {
				g.log.Errorf("checking for cloud events: %v", err)
			}```

Comment (aldor007) [reply handler/handler.go:95]: Missed that :)

**Review by andrejatcastai — APPROVED** [review]

**Pull request merged**